### PR TITLE
Jit64Common: Split Jit64Util contents into separate files

### PR DESCRIPTION
**Pull request by lioncash**
Much of Jit64Util consists of essentials, not utilities. Breaking these out into their own files also prevents unrelated includes from being present near other classes.

This also makes it easier to find and change certain components of the x86-64 JIT, should it be necessary.

This is essentially a preliminary cleanup before I make a few of the memory routines for Jit64 not rely on the `jit` global variable. Eliminating reliance on the global will also allow me to rework #4518 without the need for downcasting.

### CODE REVIEW

**Comment by iwubcode** [comment]
Looks good, should Jit64Util.h be renamed to something more appropriate like PowerPCState?

**Comment by lioncash** [comment]
> Looks good, should Jit64Util.h be renamed to something more appropriate like PowerPCState?

PowerPCState is a thing that already [exists](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/PowerPC/PowerPC.h#L59) in an unrelated header. So I've left it as Jit64Utils.

**Comment by iwubcode** [comment]
Jit64PPCStates then?  Jit64Util doesn't really tell me a whole lot.

What exactly are those defines for?

**Comment by lioncash** [comment]
They generate an operand argument that uses RPPCState (an actual x86 register for tracking the live PowerPCState) and contains the offset to the relevant entity in the PowerPCState struct.

**Comment by iwubcode** [comment]
Ah ok, thanks for the explanation.  Well, if you want to leave it that's fine, I just feel like Util doesn't really explain the file very well.  Something like PPCStateOffsets or something would make more sense to me.

Otherwise LGTM from me.

File: Source/Core/Core/PowerPC/Jit64Common/Jit64PowerPCState.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Core/PowerPC/PowerPC.h"
+
+// We offset by 0x80 because the range of one byte memory offsets is
+// -0x80..0x7f.
+#define PPCSTATE(x)                                                                                \
+  MDisp(RPPCSTATE, (int)((char*)&PowerPC::ppcState.x - (char*)&PowerPC::ppcState) - 0x80)
```
Comment (BhaaLseN) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64PowerPCState.h]: Aren't those candidates for cpp-style casts?

Comment (iwubcode) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64PowerPCState.h]: I agree, not sure what @lioncash  will say but I personally think that would be better off for a separate commit, as the original code also had that issue.

Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64PowerPCState.h]: Yep, this commit is mostly for shuffling around existing code.

**Comment by lioncash** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/PowerPC/Jit64Common/Jit64PowerPCState.h (hunk 1 above)
Comment (BhaaLseN) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64PowerPCState.h]: Thought so too afterwards, so yeah, LGTM.
